### PR TITLE
[Feature](auditloader) Plugin auditloader use auth token to avoid using cleartext passwords in config

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
1. Doris FE will check if stream load http request has auth token after checking password failed;
2. Plugin audit-log loader can use auth token if plugin config set use_auth_token to true

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 1]
```diff
@@ -257,4 +269,95 @@ private TNetworkAddress selectRedirectBackend(String clusterName) throws LoadExc
...
+                redirectUrl += request.getQueryString();
+            }
+            LOG.info("Redirect url: {}", redirectUrl);
+            RedirectView redirectView = new RedirectView(redirectUrl);
+            redirectView.setContentType("text/html;charset=utf-8");
+            redirectView.setStatusCode(org.springframework.http.HttpStatus.TEMPORARY_REDIRECT);
+
+            return redirectView;
+        } catch (Exception e) {
+            return new RestBaseResult(e.getMessage());
```
Comment (CalvinKirs) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:360]: This will lose exception stack information,

Comment (zhiqiang-hhhh) [reply fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:362]: Fixed

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 2]
```diff
@@ -257,4 +269,97 @@ private TNetworkAddress selectRedirectBackend(String clusterName) throws LoadExc
         }
         return new TNetworkAddress(backend.getHost(), backend.getHttpPort());
     }
+
+    // NOTE: This function can only be used for AuditlogPlugin stream load for now.
+    // AuditlogPlugin should be re-disigned carefully, and blow method focuses on
+    // temporarily addressing the users' needs for audit logs.
+    // So this function is not widely tested under general scenario
+    private boolean checkClusterToken(HttpServletRequest request) {
+        LOG.info("Checking cluser token, request {}", request.toString());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:278]:
```suggestion
        LOG.debug("Checking cluser token, request {}", request.toString());
```

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 3]
```diff
@@ -257,4 +269,97 @@ private TNetworkAddress selectRedirectBackend(String clusterName) throws LoadExc
...
+        String authToken = request.getHeader("token");
+
+        if (Strings.isNullOrEmpty(authToken)) {
+            LOG.warn("Failed to check cluster token.");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:282]: I think this is not a warn.
In normal case, if user submit load with wrong username/password, the `authToken` is always empty.
So I think we can remove this log

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 4]
```diff
@@ -257,4 +269,97 @@ private TNetworkAddress selectRedirectBackend(String clusterName) throws LoadExc
...
+                redirectUrl += request.getQueryString();
+            }
+            LOG.info("Redirect url: {}", redirectUrl);
+            RedirectView redirectView = new RedirectView(redirectUrl);
+            redirectView.setContentType("text/html;charset=utf-8");
+            redirectView.setStatusCode(org.springframework.http.HttpStatus.TEMPORARY_REDIRECT);
+
+            return redirectView;
+        } catch (Exception e) {
+            LOG.error("Failed to execute stream load with cluster token, {}", e);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:361]:
```suggestion
            LOG.warn("Failed to execute stream load with cluster token, {}", e);
```

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java (hunk 3 above)
Comment (zhiqiang-hhhh) [reply fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java]: Fixed

**Comment by zhiqiang-hhhh** [comment]
@morningman @CalvinKirs

**Review by morningman — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**